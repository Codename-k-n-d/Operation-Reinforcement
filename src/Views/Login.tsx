import React from "react";

function Login(): JSX.Element {
  return (
    <div>
      <h1>This is Login</h1>
      <form>
        <input className='username-input'/>
        <input className='password-input'/>
        <input className='submit-button' type='submit'/>
      </form>
    </div>
  )
}

export default Login;