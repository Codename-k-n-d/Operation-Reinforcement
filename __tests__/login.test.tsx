//require in login page
import React from 'react';
import TestRenderer from 'react-test-renderer';
import { create } from 'react-test-renderer';
import Login from '../src/Views/Login';

describe('<Login />', () => {
  const testRenderer = TestRenderer.create(<Login />);
  // snapshot testing
  it("matches the snapshot", () => {
    // @ts-ignore
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
/*   const testInstance = testRenderer.root; 
  it('username field should render placeholder text', () => {
    console.log(testInstance);
    //expect(testInstance.props.className.includes('username-input')).toBe(true);
  });
  it('password field should render placeholder text', () => {

  });
  it('renders a button', () => {

  }); */
})


function Button() {
  return <button>Nothing to do for now</button>;
}

describe('Button Component', () => {
  it("Matches the snapshot", () => {
    const button = TestRenderer.create(<Button />);
    expect(button.toJSON()).toMatchSnapshot();
  });
  it
});

