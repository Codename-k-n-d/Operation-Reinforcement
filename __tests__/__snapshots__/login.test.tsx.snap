// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> matches the snapshot? 1`] = `
<div>
  <h1>
    This is Login
  </h1>
  <form>
    <input
      className="username-input"
    />
    <input
      className="password-input"
    />
    <input
      className="submit-button"
      type="submit"
    />
  </form>
</div>
`;

exports[`Button Component Matches the snapshot 1`] = `
<button>
  Nothing to do for now
</button>
`;
